browserpass: init at 1.0.2

pajowu opened the pull request:
###### Motivation for this change

The browser-extension for pass (browserpass) needs a native component. That component is installed using this package. The Config-files are added to chrome and chromium with the `programs.browserpass` module.

###### Things done

- [X] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [x] macOS
   - [x] Linux
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

pajowu committed:
browserpass: init at 1.0.2

danielfullmer commented:
I have also been using this very useful chrome extension in my own nixos configuration, but hadn't packaged it up as a module like in this pull request. If it's useful, see these as well:
https://github.com/danielfullmer/nixrc/blob/master/pkgs/browserpass/default.nix
and
https://github.com/danielfullmer/nixrc/blob/master/profiles/desktop/default.nix#L197

@pajowu  Instead of using the prebuilt binary from upstream, I'd recommend using `buildGoPackage` against the source code.

pajowu commented:
@danielfullmer that looks great, i'll try to change it to build from git. That way it would also support 32bit

pajowu committed:
browserpass: switch build source to git HEAD

Precompiled version only supports linux64, git version should also support darwin

jagajaga commented:
Why not just https://github.com/nwallace/passff? :)

jagajaga commented on pkgs/tools/security/browserpass/default.nix:
```diff
@@ -0,0 +1,34 @@
+
+buildGoPackage rec {
+  name = "browserpass-unstable-${version}";
+  version = "HEAD";
```
Use proper way to name pkg. In this case it should be `browserpass-(date of commit)`. Also please use `sha` for consistency.

pajowu replied:
fixed with 11913f2ce1

pajowu committed:
browserpass: change naming scheme

pajowu commented:
@jagajaga 

> Why not just https://github.com/nwallace/passff? :)

passff also needs a native extension, which would need a similar install method. Also giving more options seems like a good thing

pajowu committed:
browserpass: add support for firefox

7c6f434c committed:
Remove empty longDescription

wucke13 commented:
@ttuegel @pajowu 
This is still not resolved, browserpass does not work for me on the most recent firefox package in NixOS 17.09
